pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -2719,8 +2761,123 @@ int BPF_KPROBE(trace_security_socket_connect)
         stsb(args_buf, (void *) address, sockaddr_len, 2);
     }
 
-    // Submit the event.
-    return events_perf_submit(&p, SECURITY_SOCKET_CONNECT, 0);
+    return events_perf_submit(p, SECURITY_SOCKET_CONNECT, 0);
+}
+
+statfunc int send_stdio_via_socket_from_sock_connect(struct pt_regs *ctx, program_data_t *p) {
+    u64 addr_len = PT_REGS_PARM3(ctx);
+
+    struct socket *sock = (struct socket *) PT_REGS_PARM1(ctx);
+    if (!sock)
+        return 0;
+
+    struct sockaddr *address = (struct sockaddr *) PT_REGS_PARM2(ctx);
+    if (!address)
+        return 0;
+
+    // Check if the socket type is supported.
+    u32 type = BPF_CORE_READ(sock, type);
+    switch (type) {
+        // TODO: case SOCK_DCCP:
+        case SOCK_DGRAM:
+        case SOCK_SEQPACKET:
+        case SOCK_STREAM:
+            break;
+        default:
+            return 0;
+    }
+
+    // Check if the socket family is supported.
+    sa_family_t sa_fam = get_sockaddr_family(address);
+    switch (sa_fam) {
+        case AF_INET:
+        case AF_INET6:
+        default:
+            return 0;
+    }
+
+    // Load args given to the syscall that invoked this function.
+    syscall_data_t *sys = &p->task_info->syscall_data;
+    if (!p->task_info->syscall_traced)
+        return 0;
+
+    // Reduce line cols by having a few temp pointers.
+    reset_event_args(p);
+    int (*stsb)(args_buffer_t *, void *, u32, u8) = save_to_submit_buf;
+    void *args_buf = &p->event->args_buf;
+    void *to = (void *) &sys->args.args[0];
+
+    if (is_x86_compat(p->task)) // only i386 binaries uses socketcall
+        to = (void *) sys->args.args[1];
+
+    if (!is_stdio_via_socket((u64)to, sa_fam)) {
```

Comment:
This `if` is effectively the only difference from the function above. I am not sure if this is such a good idea, as we are fairly limited with stack size, as well as instruction count in eBPF. 

With which kernel did you test the changes?

Reply:
Stack size for ebpf program is 512 bytes. If I would move it to separate tail calls we would get 256 bytes for each program.

Options:
1. Leave it as is and refactor if limit is reached. I'm testing on 5.15 kernel.
2. Move to separate tail calls.
3. Create separate named kprobes for each case.
4. Remove code which we don't need. In our case we will not output security socket connect to user space anyway. Maybe it's not worth maintaining that event.

Reply:
Wait, if you move it to a tail call, you should get the full `512bytes` 🤔

Wouldn't it be enough to move 
```
if (!is_stdio_via_socket((u64)to, sa_fam)) {
```
to the end of the function above? 

I mean option 4 would also be valid. If we in the future decide that we want to have the event again, adding it is not that big of a deal 👌